staging/src/k8s.io/dynamic-resource-allocation/structured/internal/stable/pools_stable.go

```diff
@@ -181,6 +148,43 @@ func addSlice(pools map[PoolID]*Pool, s *resourceapi.ResourceSlice) error {
 	return nil
 }
 
+// findSlicesForPool returns all ResourceSlices that belong to the provided pool.
+func findSlicesForPool(slices []*resourceapi.ResourceSlice, poolID PoolID) []*resourceapi.ResourceSlice {
```

Comment:
Haven't we already determined the desired generation? Then we can limit collecting slices to that generation and the "pool is incomplete" check becomes simpler.

Reply:
I added another commit that I think addresses this. I also added a test to make sure devices from stale ResourceSlices does not get allocated.

Reply:
https://github.com/kubernetes/kubernetes/pull/134105 added to [my implementation](https://github.com/kubernetes/kubernetes/pull/135036/files#r2485557692) is slightly better, but still not as fast as before (-51.05%).

Reply:
See https://github.com/kubernetes/kubernetes/pull/135056 for an alternative for this PR which includes my thoughts on "detect more recent generation".